azuredevops_git_permissions is crashing when changing the provider organization URL

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and Azure DevOps Provider) Version

```
$ terraform version
Terraform v1.11.4
on windows_amd64
+ provider registry.terraform.io/microsoft/azuredevops v1.11.2
```

### Affected Resource(s)

* `azuredevops_git_permissions`

### Terraform Configuration Files

This is a small reproducible test case:

```hcl
terraform {
  required_providers {
    azuredevops = {
      source  = "microsoft/azuredevops"
      version = "~> 1.11.2"
    }
  }
}
variable "organization_name" {
  type    = string
  default = "organizationA"
}
variable "project_name" {
  type    = string
  default = "projectA"
}
variable "repository_name" {
  type    = string
  default = "repositoryA"
}
variable "team_name" {
  type = string
  default = "teamA"
}
provider "azuredevops" {
  org_service_url = "https://dev.azure.com/${var.organization_name}"
}
data "azuredevops_project" "standard" {
  name = var.project_name
}
resource "azuredevops_team" "team" {
  project_id = data.azuredevops_project.standard.id
  name       = var.team_name
}
resource "azuredevops_git_repository" "repository" {
  project_id     = data.azuredevops_project.standard.id
  name           = var.repository_name
  default_branch = "refs/heads/main"
  initialization {
    init_type = "Clean"
  }
  lifecycle {
    ignore_changes = [
      initialization,
    ]
  }
}
resource "azuredevops_git_permissions" "repository_permissions" {
  project_id    = data.azuredevops_project.standard.id
  repository_id = azuredevops_git_repository.repository.id
  principal     = azuredevops_team.team.descriptor
  permissions = {
    GenericRead           = "Allow"
    GenericContribute     = "Allow"
    PullRequestContribute = "Allow"
  }
}
```

### Debug Output

```
│ Error: Plugin did not respond
│
│   with azuredevops_git_permissions.repository_permissions,
│   on main.tf line 52, in resource "azuredevops_git_permissions" "repository_permissions":
│   52: resource "azuredevops_git_permissions" "repository_permissions" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-azuredevops_v1.11.2 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xeebf38]

goroutine 49 [running]:
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils.(*SecurityNamespace).GetPrincipalPermissions.func1({0x1302c60?, 0xc000783950?})
        github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils/namespaces.go:455 +0x98
reflect.Value.call({0x1148940?, 0x1390df8?, 0xc0007d6ab8?}, {0x132d176, 0x4}, {0xc0004684b0, 0x1, 0x1c4fe2c1368?})
        reflect/value.go:584 +0xca6
reflect.Value.Call({0x1148940?, 0x1390df8?, 0xc000783050?}, {0xc0004684b0?, 0xc000468498?, 0xc0007d6b48?})
        reflect/value.go:368 +0xb9
github.com/ahmetb/go-linq.(*genericFunc).Call(0xc00051a0b0, {0xc0007d6b60, 0x1, 0x1?})
        github.com/ahmetb/go-linq@v3.0.0+incompatible/genericfunc.go:35 +0x107
github.com/ahmetb/go-linq.Query.SelectT.func1({0x1302c60?, 0xc000783950?})
        github.com/ahmetb/go-linq@v3.0.0+incompatible/select.go:47 +0x3b
github.com/ahmetb/go-linq.Query.SelectT.Query.Select.func3.1()
        github.com/ahmetb/go-linq@v3.0.0+incompatible/select.go:24 +0x3b
github.com/ahmetb/go-linq.Query.ToSlice({0x1111740?}, {0x110c5c0?, 0xc000468438?})
        github.com/ahmetb/go-linq@v3.0.0+incompatible/result.go:634 +0x1be
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils.(*SecurityNamespace).GetPrincipalPermissions(0xc00050e3c0, 0xc0007d7010)
        github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils/namespaces.go:457 +0xd3
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils.GetPrincipalPermissions(0xc0004ee800, 0xc00050e3c0)
        github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils/securityHelper.go:107 +0x105
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions.resourceGitPermissionsRead(0xc0004ee800, {0x10e2e00?, 0xc0001c61e0?})
        github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/resource_git_permissions.go:77 +0x72
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x14c45e8?, {0x14c45e8?, 0xc0007277a0?}, 0xd?, {0x10e2e00?, 0xc0001c61e0?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:843 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0003a4400, {0x14c45e8, 0xc0007277a0}, 0xc0004584e0, {0x10e2e00, 0xc0001c61e0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:1149 +0x53a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00036c798, {0x14c45e8?, 0xc0007276e0?}, 0xc000416240)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:739 +0x6c5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000279720, {0x14c45e8?, 0xc000726f00?}, 0xc0000c0150)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:784 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1300aa0, 0xc000279720}, {0x14c45e8, 0xc000726f00}, 0xc0004ee000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:575 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00027e400, {0x14c45e8, 0xc000726ea0}, 0xc00050e1e0, 0xc0003f3b60, 0x1c1b7c8, 0x0)
        google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc00027e400, {0x14c4f18, 0xc000492680}, 0xc00050e1e0)
        google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 16
        google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-azuredevops_v1.11.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Panic Output

No panic

### Expected Behavior

The new resource should be created without issue

### Actual Behavior

The provider crashes (see above for the output)

### Steps to Reproduce

1. `terraform init`
1. `terraform apply`
1. `TF_VAR_organization_name=organizationB TF_VAR_project_name=projectB terraform apply`

### Important Factoids


### References

--- a/azuredevops/internal/service/permissions/utils/namespaces.go
+++ b/azuredevops/internal/service/permissions/utils/namespaces.go
@@ -244,7 +244,7 @@ func (sn *SecurityNamespace) GetAccessControlList(descriptorList *[]string) (*se
 		return nil, nil
 	}
 
-	var descriptors *string = nil
+	var descriptors *string
 	val := linq.From(*descriptorList).
 		Aggregate(func(r interface{}, i interface{}) interface{} {
 			if r.(string) == "" {

--- a/azuredevops/internal/service/permissions/utils/namespaces.go
+++ b/azuredevops/internal/service/permissions/utils/namespaces.go
@@ -240,17 +240,19 @@ func (sn *SecurityNamespace) GetActionDefinitions() (*map[string]security.Action
 }
 
 func (sn *SecurityNamespace) GetAccessControlList(descriptorList *[]string) (*security.AccessControlList, error) {
-	var descriptors *string = nil
-	if descriptorList != nil && len(*descriptorList) > 0 {
-		val := linq.From(*descriptorList).
-			Aggregate(func(r interface{}, i interface{}) interface{} {
-				if r.(string) == "" {
-					return i
-				}
-				return r.(string) + "," + i.(string)
-			}).(string)
-		descriptors = &val
+	if descriptorList == nil || len(*descriptorList) == 0 {
+		return nil, nil
 	}
+
+	var descriptors *string = nil
+	val := linq.From(*descriptorList).
+		Aggregate(func(r interface{}, i interface{}) interface{} {
+			if r.(string) == "" {
+				return i
+			}
+			return r.(string) + "," + i.(string)
+		}).(string)
+	descriptors = &val
 
 	bTrue := true
 	acl, err := sn.securityClient.QueryAccessControlLists(sn.context, security.QueryAccessControlListsArgs{
@@ -269,6 +271,37 @@ func (sn *SecurityNamespace) GetAccessControlList(descriptorList *[]string) (*se
 		return nil, fmt.Errorf("Failed to load current ACL for token [%s]. Result set contains more than one ACL", sn.token)
 	}
 	return &(*acl)[0], nil
+}
+
+func (sn *SecurityNamespace) tryGetIdentitiesFromSubjects(principal *[]string) (*[]identity.Identity, error) {
+	descriptors := linq.From(*principal).
+		Aggregate(func(r interface{}, i interface{}) interface{} {
+			if r.(string) == "" {
+				return i
+			}
+			return r.(string) + "," + i.(string)
+		}).(string)
+
+	idlist, err := sn.identityClient.ReadIdentities(sn.context, identity.ReadIdentitiesArgs{
+		SubjectDescriptors: converter.String(descriptors),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var emptyId identity.Identity
+	linq.From(*idlist).Where(func(ele interface{}) bool {
+		if val, ok := ele.(identity.Identity); ok {
+			if val == emptyId {
+				return false
+			}
+			if val.IsActive == nil || val.IsActive != nil && *val.IsActive {
+				return true
+			}
+		}
+		return false
+	}).ToSlice(idlist)
+	return idlist, nil
 }
 
 func (sn *SecurityNamespace) getIdentitiesFromSubjects(principal *[]string) (*[]identity.Identity, error) {
@@ -295,8 +328,12 @@ func (sn *SecurityNamespace) getIdentitiesFromSubjects(principal *[]string) (*[]
 		return nil, fmt.Errorf("No identity information for defined principals [%s]", descriptors)
 	}
 
+	var emptyId identity.Identity
 	linq.From(*idlist).Where(func(ele interface{}) bool {
 		if val, ok := ele.(identity.Identity); ok {
+			if val == emptyId {
+				return false
+			}
 			if val.IsActive == nil || val.IsActive != nil && *val.IsActive {
 				return true
 			}
@@ -444,7 +481,7 @@ func (sn *SecurityNamespace) GetPrincipalPermissions(principal *[]string) (*[]Pr
 		return nil, err
 	}
 
-	idList, err := sn.getIdentitiesFromSubjects(principal)
+	idList, err := sn.tryGetIdentitiesFromSubjects(principal)
 	if err != nil {
 		return nil, err
 	}

